trie/sync_bloom.go

```diff
@@ -171,11 +157,11 @@ func (b *SyncBloom) Close() error {
 }
 
 // Add inserts a new trie node hash into the bloom filter.
-func (b *SyncBloom) Add(hash []byte) {
+func (b *SyncBloom) Add(hash common.Hash) {
```

Comment:
I'm unsure this `[]byte` to `common.Hash` change is a good idea. The old code throughout either used some existing `[]byte` value, or if it has a hash, it did `hash[:]`. In both these cases, there's no copy involved as we're jut using the same backing memory area.

With this refactor however, you are making the parameter an array from potentially a `[]byte`, which always incurs a full copy.

Reply:
The `SyncBloom.Add` always takes a `common.Hash` without conversion, afaict